app/lib/actions/proxy_action.rb

```diff
@@ -95,12 +111,21 @@ def on_resume
     # @override to put custom logic on event handling
     def on_data(data)
       output[:proxy_output] = data
-      wipe_secrets!
     end
 
-    def wipe_secrets!
+    # Removes the :secrets key from the action's input and output and saves the action
+    def wipe_secrets!()
       input.delete(:secrets)
       output.delete(:secrets)
+      world.persistence.save_action(execution_plan_id, self)
+    end
+
+    def on_proxy_action_missing
+      error! _('Proxy task gone missing from the smart proxy')
```

Comment:
I guess we should use this method also in `check_task_status`, in case the remote task is missing. The reason:

1. I schedule the job
2. restart the smart proxy dynflow before the connection timeout happens 
3. during the connection timeout, the task fails with 'Error loading data from proxy: RestClient::NotFound - 404 Not Found', instead of saying that "'Proxy task gone missing from the smart proxy'".

I wonder if the `foreman_tasks_proxy_action_start_timeout` actually still makes sense: it would free up som unnecessary events we are emoting at the moment

Reply:
I dropped the `foreman_tasks_proxy_action_start_timeout` completely